erts/emulator/beam/bif.c

```diff
@@ -4417,7 +4417,7 @@ BIF_RETTYPE make_fun_3(BIF_ALIST_3)
     }
 
     arity = signed_val(BIF_ARG_3);
-    if (arity < 0) {
+    if (arity < 0 || arity > MAX_ARG) {
```

Comment:
You might get better code from `if ((Uint)arity > MAX_ARG)`

Reply:
Thanks for pointing it out, but I think it's clearer this way and I trust the compiler to do that optimization on its own.